src/sidebar/util/type-coercions.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Type conversion methods that gently coerce values that other parts
+ * of the UI may make assumptions on. This is needed for incoming parameters
+ * that are otherwise not sanitized. Note that these methods do not throw 
+ * errors or act like validators with mal-formed values.
+ */
+
+/**
+ * Returns a boolean
+ *
+ * @param {any} value - initial value
+ * @return {boolean}
+ */
+function toBoolean(value) {
+  if (typeof value === 'string') {
+    if (value.trim().toLocaleLowerCase() === 'false') {
+      // "false", "False", " false", "FALSE" all return false
+      return false;
+    }
+  }
+  const numericalVal = Number(value);
+  if (!isNaN(numericalVal)) {
+    return Boolean(numericalVal);
+  }
+  // Any non numerical or falsely string should return true, otherwise
+  // return false
+  return typeof value === 'string';
+}
+
+/**
+ * Returns either an integer or NaN
+ *
+ * @param {any} value - initial value
+ * @return {number}
+ */
+function toInteger(value) {
+  // Acts as a simple wrapper
+  return parseInt(value);
+}
+
+/**
+ * Returns either the value if its an object or an empty object
+ *
+ * @param {any} value - initial value
+ * @return {object}
+ */
+function toObject(value) {
+  if (typeof value === 'object' && value !== null) {
+    return value;
+  }
+  // Don't attempt to fix the values, just ensure type correctness
+  return {};
+}
+
+/**
+ * Returns the value as a string or an empty string if the
+ * value undefined or null.
+ *
+ * @param {any} value - initial value
+ * @return {string}
+ */
+function toString(value) {
+  if (value && value.toString) {
+    return value.toString();
```

Comment:
This could throw if `toString` is not a function. Assuming the incoming value is produced by `JSON.parse`, this could throw if given `{ toString: false }`.

```js
if (!value) {
  return '';
}
try {
  return String(value);
} catch (e) {
  return '';
}
```